src/core.e/PseudoStash.pm6

```diff
@@ -480,7 +483,7 @@ my class PseudoStash is Map {
                             ($sym := nqp::iterkey_s($!iter)),
                             # The symbol has to be dynamic if pseudo-package is marked as requiring dynamics or if
                             # we'recurrently iterating over the dynamic chain.
-                            ($got-one := !nqp::atkey($!seen,$sym) && (
+                            ($got-one := !nqp::defined(nqp::atkey($!seen,$sym)) && (
```

Comment:
Well, AFAIK nqp::atkey returns null for the 'not exists' case on the JVM backend. And trying to negate that null results in a NullPointerException:
```
$ ./perl6-j -e 'use nqp; my $foo := nqp::hash(); my $bar := !nqp::atkey($foo,"foo")'
java.lang.NullPointerException
  in block <unit> at -e line 1
```

Just adding the nqp::defined looks inprecise, though. Reading your comment ("Values in ```$!seen``` would either exists and be 1 or not exists) and the code of next-one() again, wouldn't using nqp::existskey instead of nqp::atkey suffice:
```
($got-one := !nqp::existskey($!seen,$sym) && ( ...
```

Reply:
I was thinking about `nqp::existskey` it too. Don't think it would ever be necessary to store `0`s in `$!seen`. I'd be ok with it.

But `!nqp::null()` failing on JVM sounds like a bug to me. Could be a source of many other problems around the core.

Reply:
> But !nqp::null() failing on JVM sounds like a bug to me. Could be a source of many other problems around the core.

That's very true. For the record, it seems to work with NQP:
```
$ (cd nqp; ./nqp-j -e 'say(!nqp::null)')
1
$ ./perl6-j -e 'use nqp; nqp::say(!nqp::null)'
java.lang.NullPointerException
  in block <unit> at -e line 1
```

Reply:
This should be fixed with https://github.com/perl6/nqp/commit/670400e625

No need for ```s/atkey/existskey/```.